go/client/chat_svc_handler.go

```diff
@@ -281,6 +282,39 @@ func (c *chatServiceHandler) SetUnfurlSettingsV1(ctx context.Context, opts setUn
 	return Reply{Result: true}
 }
 
+func (c *chatServiceHandler) GetDeviceInfoV1(ctx context.Context, opts getDeviceInfoOptionsV1) Reply {
+	client, err := GetUserClient(c.G())
+	if err != nil {
+		return c.errReply(err)
+	}
+	user, err := client.LoadUserByName(ctx, keybase1.LoadUserByNameArg{Username: opts.Username})
```

Comment:
is LoadUserByName the best way to get a Uid from a Username?

Reply:
It's ok. [LoadUserByName](https://github.com/keybase/client/blob/master/go/service/user.go#L93) is just really old so it does a full user load. It could probably be updated to use CachedUPAKLoader.

Imo the cleanest way for this to work would be if the work happens on the service, and this client method is just a pass through to a local RPC.

Loading from CachedUPAKLoader looks like this.
```
upak, _, err = mctx.G().GetUPAKLoader().LoadV2(
  libkb.NewLoadUserArgWithMetaContext(mctx).WithPublicKeyOptional()
  .WithName(username))
check(err)
upak.Current
```

Any of these is fine by me
1. Leave this as is.
2. Update LoadUserByName to use UPAKLoader to lookup username->uid. Which could also speed up old code. But is also a tiny bit destabilizing.
3. Move GetDeviceInfo into a new service handler which can use UPAKLoader directly.